orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/manifest/WaitForManifestStableTask.java

```diff
@@ -55,6 +57,20 @@ public long getTimeout() {
     return TimeUnit.MINUTES.toMillis(30);
   }
 
+  @Override
+  public long getDynamicTimeout(StageExecution stage) {
+    ManifestTimeout timeout = stage.mapTo(ManifestTimeout.class);
+    if (timeout.timeoutMinutes != null && timeout.timeoutMinutes.isPresent()) {
```

Comment:
I don't think there's much benefit to using an `Optional` here if we still need to check whether it's null.  It looks like Jackson will set `timeoutMinutes` to `Optional.empty()` if the field `timeoutMinutes` is present but is set to something it considers empty (ex: null or empty string), whereas it will just leave it as `null` if the field is entirely absent from the input.  I think that `ManifestTimeout` should handle all this complexity and let callers assume that `timeoutMinutes` is non-null.

I think the best way of doing that is to change the `ManifestTimeout` class to something like this:
```
    @Nonnull private final OptionalLong timeoutMinutes;

    ManifestTimeout(@JsonProperty("timeoutMinutes") OptionalLong timeoutMinutes) {
      this.timeoutMinutes = timeoutMinutes;
    }
```

This will guarantee that Jackson will pass an `OptionalLong.empty()` to the constructor if the field is not present, as opposed to using setter-based construction which will only set fields that are present.  It also has the advantage of making the class immutable.  Then you can remove the `== null` check and only do the `.isPresent` check.

I think there should be another test of what happens if `timeoutMinutes` is present but set to an empty string (as would probably happen if it was left unset in the UI).  I would also suggest adding a test where `timeoutMinutes` is set to the string `"1"` as I believe that will be what comes back from deck in this case.

With those additional tests we should be in pretty good shape around making sure this works for all cases (and even if something like a Jackson upgrade breaks it, we'll find out right away).